aviatrix/resource_aviatrix_transit_gateway.go

```diff
@@ -1871,42 +1871,70 @@ func resourceAviatrixTransitGatewayRead(d *schema.ResourceData, meta interface{}
 
 	// edge cloud type
 	if goaviatrix.IsCloudType(gw.CloudType, goaviatrix.EdgeRelatedCloudTypes) {
-		d.Set("device_id", gw.DeviceID)
 		d.Set("site_id", gw.SiteID)
-		// TODO: should this be b64 encoded? set the interfaces from gw.Interfaces
+		d.Set("bgp_lan_ip_list", nil)
+		d.Set("ha_bgp_lan_ip_list", nil)
 		if len(gw.Interfaces) != 0 {
 			var interfaces []map[string]interface{}
-			for _, intf := range gw.Interfaces {
+			sortedInterfaces := sortInterfacesByCustomOrder(gw.Interfaces)
+			for _, intf := range sortedInterfaces {
 				interfaceDict := make(map[string]interface{})
-				interfaceDict["ifname"] = intf.IfName
+				interfaceDict["name"] = intf.Name
 				interfaceDict["type"] = intf.Type
 				if intf.PublicIp != "" {
 					interfaceDict["public_ip"] = intf.PublicIp
 				}
 				if intf.Dhcp {
 					interfaceDict["dhcp"] = intf.Dhcp
 				}
-				if intf.IpAddr != "" {
-					interfaceDict["ipaddr"] = intf.IpAddr
+				if intf.IpAddress != "" {
+					interfaceDict["ip_address"] = intf.IpAddress
 				}
 				if intf.GatewayIp != "" {
 					interfaceDict["gateway_ip"] = intf.GatewayIp
 				}
-				if len(intf.SecondaryCIDRs) > 0 {
-					interfaceDict["secondary_private_cidr_list"] = intf.SecondaryCIDRs
+				if intf.SecondaryCIDRs != nil {
+					secondaryCIDRs := make([]string, 0)
+					for _, cidr := range intf.SecondaryCIDRs {
+						if cidr != "" {
+							secondaryCIDRs = append(secondaryCIDRs, cidr)
+						}
+					}
+					interfaceDict["secondary_private_cidr_list"] = secondaryCIDRs
 				}
 				interfaces = append(interfaces, interfaceDict)
+				log.Printf("[TRACE] Interface Dictionary %s: %#v", intf.Name, interfaceDict)
 			}
 			if err = d.Set("interfaces", interfaces); err != nil {
 				return fmt.Errorf("could not set interfaces into state: %v", err)
 			}
 		}
-		// set the interface map
-		d.Set("interface_map", gw.InterfaceMapping)
-		if gw.PeerBackupPort != "" && gw.ConnectionType != "" {
-			d.Set("peer_backup_port", gw.PeerBackupPort)
-			d.Set("connection_type", gw.ConnectionType)
```

Comment:
trying to understand, here you just no longer set the interface map?

Reply:
Interface map is being set statically in the code and so I not setting it in the EAT state here.